### ISSUE TITLE
Add JMX metrics encountered a bind failed

### ISSUE THREAD

**Issue by maobaolong** [Alluxio/alluxio#15168]
**Alluxio Version:**

Master branch, commit hash 2f0c99044dfd7f8578bd3cc5606b4c0ef5b8a7a7

**Describe the bug**
```
Error: Exception thrown by the agent : java.rmi.server.ExportException: Port already in use: 1102; nested exception is: 
        java.net.BindException: Address already in use (Bind failed)
sun.management.AgentConfigurationError: java.rmi.server.ExportException: Port already in use: 1102; nested exception is: 
        java.net.BindException: Address already in use (Bind failed)
        at sun.management.jmxremote.ConnectorBootstrap.startRemoteConnectorServer(ConnectorBootstrap.java:480)
        at sun.management.Agent.startAgent(Agent.java:262)
        at sun.management.Agent.startAgent(Agent.java:452)
Caused by: java.rmi.server.ExportException: Port already in use: 1102; nested exception is: 
        java.net.BindException: Address already in use (Bind failed)
        at sun.rmi.transport.tcp.TCPTransport.listen(TCPTransport.java:346)
        at sun.rmi.transport.tcp.TCPTransport.exportObject(TCPTransport.java:254)
        at sun.rmi.transport.tcp.TCPEndpoint.exportObject(TCPEndpoint.java:412)
        at sun.rmi.transport.LiveRef.exportObject(LiveRef.java:147)
        at sun.rmi.server.UnicastServerRef.exportObject(UnicastServerRef.java:237)
        at sun.rmi.registry.RegistryImpl.setup(RegistryImpl.java:213)
        at sun.rmi.registry.RegistryImpl.<init>(RegistryImpl.java:173)
        at sun.management.jmxremote.SingleEntryRegistry.<init>(SingleEntryRegistry.java:49)
        at sun.management.jmxremote.ConnectorBootstrap.exportMBeanServer(ConnectorBootstrap.java:816)
        at sun.management.jmxremote.ConnectorBootstrap.startRemoteConnectorServer(ConnectorBootstrap.java:468)
        ... 2 more
Caused by: java.net.BindException: Address already in use (Bind failed)
        at java.net.PlainSocketImpl.socketBind(Native Method)
        at java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:387)
        at java.net.ServerSocket.bind(ServerSocket.java:390)
        at java.net.ServerSocket.<init>(ServerSocket.java:252)
        at java.net.ServerSocket.<init>(ServerSocket.java:143)
        at sun.rmi.transport.proxy.RMIDirectSocketFactory.createServerSocket(RMIDirectSocketFactory.java:45)
        at sun.rmi.transport.proxy.RMIMasterSocketFactory.createServerSocket(RMIMasterSocketFactory.java:345)
        at sun.rmi.transport.tcp.TCPEndpoint.newServerSocket(TCPEndpoint.java:670)
        at sun.rmi.transport.tcp.TCPTransport.listen(TCPTransport.java:335)
        ... 11 more
```
**To Reproduce**

Add the following content into `alluxio-env.sh` and execute `bin/alluxio-start.sh master`

```
ALLUXIO_MASTER_JAVA_OPTS="-Dcom.sun.management.jmxremote \
-Dcom.sun.management.jmxremote.port=1102 \
-Dcom.sun.management.jmxremote.local.only=false \
-Dcom.sun.management.jmxremote.authenticate=false \
-Dcom.sun.management.jmxremote.ssl=false"
```

**Expected behavior**

Master should started successfully, but report `Address already in use`

**Urgency**

High

**Are you planning to fix it**

No

**Additional context**
No

**Comment by jiacheliu3** [comment]
@jenoudet I remember you fixed a similar issue when Alluxio starts and binding ports for java agent debugging. Could you take a look at this one? Could it be a similar issue?

**Comment by jenoudet** [comment]
Yes, the problem is that `ALLUXIO_MASTER_JAVA_OPTS` is passed to other processes before getting passed to the master process: to start the master process, there is the monitor process, there is also `bin/alluxio getConf`, etc... These processes bind to the port before the master process, causing the `Address already in use` problem you are seeing. 
I created the `ALLUXIO_<PROCESS>_ATTACH_OPTS` in #14991 to avoid these problems. These opts get passed only to the Alluxio process and ensure that they don't pollute the start sequence.
Thanks for the fix in #15170 @maobaolong! You can also set 
```
ALLUXIO_MASTER_ATTACH_OPTS="-Dcom.sun.management.jmxremote \
-Dcom.sun.management.jmxremote.port=1102 \
-Dcom.sun.management.jmxremote.local.only=false \
-Dcom.sun.management.jmxremote.authenticate=false \
-Dcom.sun.management.jmxremote.ssl=false"
```
and it should solve your problem.
